app/controllers/api/v1/projects_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Api::V1::ProjectsController < ApplicationController
+  include JWTHelper
+
+  skip_before_action :verify_authenticity_token
+  before_action :authenticate_jwt
+
+  # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def create
+    populate_project_from_forge
+    @project.enlistments_attributes = { '0' => { 'code_location_attributes' => { 'scm_type' => 'git',
+                                                                                 'url' => params[:url],
+                                                                                 'branch' => params[:branch] } } }
+    @project.editor_account = current_user
+    @project.assign_editor_account_to_associations
+    @project.manages.new(account: current_user) if params[:managed_by_creator].to_bool
+    @project.save!
+    create_code_location_subscription if @project.enlistments.exists?
+  rescue StandardError => e
+    render json: "Bad params #{e}", status: :bad_request
+  end
+  # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+
+  private
+
+  def project_params
+    params.require(:project).permit(
+      :name, :description, :vanity_url, :url, :download_url, :managed_by_creator, :coverity_project_id,
+      project_licenses_attributes: [:license_id],
+      enlistments_attributes: [code_location_attributes: %i[branch url scm_type]]
+    )
+  end
+
+  def populate_project_from_forge
+    match = Forge::Match.first(params[:url])
+    Timeout.timeout(Forge::Match::MAX_FORGE_COMM_TIME) { @project = match.project } if match
+  rescue Timeout::Error, OpenURI::HTTPError, URI::InvalidURIError
+    flash.now[:notice] = t('.forge_time_out', name: match.forge.name)
+  end
```

Comment:
This is similar to the function in *projects_controller*. Move both functions to a common concern and refactor this method to accept an argument for *url*.
Avoid duplicating code wherever possible.